docs/UserGuide.adoc

```diff
@@ -377,57 +412,83 @@ TODO: ADD DETAILS ONCE IMPLEMENTED
 [[FilterStudent]]
 ==== Filter `filter`
 
-TODO: ADD DETAILS ONCE IMPLEMENTED
+Filter Student returns you all the students according to the given keywords below.
```

Comment:
You can frame this better. 
For example, 

Filter student filters and returns the students according to the following keywords: